File: MRDistribuidora/mrotica_util/forms/frmSenha.vb
Imports mrotica_util
Imports System.Windows.Forms
Imports System.Drawing

Public Class frmSenha
    ' TODO: Insert code to perform custom authentication using the provided username and password 
    ' (See http://go.microsoft.com/fwlink/?LinkId=35339).  
    ' The custom principal can then be attached to the current thread's principal as follows: 
    '     My.User.CurrentPrincipal = CustomPrincipal
    ' where CustomPrincipal is the IPrincipal implementation used to perform authentication. 
    ' Subsequently, My.User will return identity information encapsulated in the CustomPrincipal object
    ' such as the username, display name, etc.
    Public usuario As New objUsuario
    Private lblNomeUsuario As New Label
    Private Sub btnEntrar_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnEntrar.Click
        Me.Close()
    End Sub
    Private Sub btnSair_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnSair.Click
        Me.Close()
    End Sub

    Private Sub frmSenha_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Dim lblUsuario As Label = New Label
        lblUsuario.Text = "Usuário"
        lblUsuario.BackColor = Color.Transparent
        lblUsuario.Location = New Point(155, 38)
        lblUsuario.Width = 50
        pictureBox1.Controls.Add(lblUsuario)

        Dim lblSenha As Label = New Label
        lblSenha.Text = "Senha"
        lblSenha.BackColor = Color.Transparent
        lblSenha.Location = New Point(242, 38)
        pictureBox1.Controls.Add(lblSenha)
    End Sub

    Private Sub txtUsuario_Leave(sender As Object, e As EventArgs) Handles txtUsuario.Leave
        If txtUsuario.Text <> "" Then
            lblNomeUsuario.Text = ""
            lblNomeUsuario.Location = New Point(155, 80)
            lblNomeUsuario.BackColor = Color.Transparent
            lblNomeUsuario.AutoSize = True
            lblNomeUsuario.ForeColor = Color.Blue
            lblNomeUsuario.Font = New Font(lblNomeUsuario.Font, FontStyle.Bold)
            pictureBox1.Controls.Add(lblNomeUsuario)
            lblNomeUsuario.Text = usuario.nome_usuario(txtUsuario.Text)
        End If
    End Sub

    Private Sub frmSenha_KeyDown(sender As Object, e As KeyEventArgs) Handles MyBase.KeyDown
        If e.KeyCode = Keys.Enter Then
            SendKeys.Send("{TAB}")
        End If
    End Sub
End Class

File: MRSystem/MRUtil/Formulario/frmSenha.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace MRUtil.Formulario
{
    public partial class frmSenha : Form
    {
        private Usuario usuario = new Usuario();
        private Label lblNomeUsuario = new Label();
        public int CodigoUsuario { get; set; }

        public int _xResposta;
        public frmSenha()
        {
            InitializeComponent();
        }

        private void frmSenha_Load(object sender, EventArgs e)
        {
            Label lblUsuario = new Label();
            lblUsuario.Text = "Usuário";
            lblUsuario.BackColor = Color.Transparent;
            lblUsuario.Location = new Point(155, 38);
            lblUsuario.Width = 50;
            pictureBox1.Controls.Add(lblUsuario);

            Label lblSenha = new Label();
            lblSenha.Text = "Senha";
            lblSenha.BackColor = Color.Transparent;
            lblSenha.Location = new Point(242, 38);
            lblSenha.Width = 50;
            pictureBox1.Controls.Add(lblSenha);
        }

        private void btnEntrar_Click(object sender, EventArgs e)
        {
            usuario.VerificaUsuarioLogado(usuario.CodigoUsuario, txtSenha.Text);
            if (usuario.UsuarioLogado == true)
            {
                CodigoUsuario = Convert.ToInt32(txtUsuario.Text);
                _xResposta = 1;
                this.Close();
            }
            else
            {
                Geral.TelaAviso("Erro ao efetuar o login. Por favor verificar se \nsuas credenciais estão corretas!", "ERROR", this, Belemtech.TipoAviso.OK, Belemtech.ImagemAviso.Error);
            }
            
        }

        private void btnSair_Click(object sender, EventArgs e)
        {
            this.Close();
            Application.Exit();
            this.Dispose();
        }

        private void txtUsuario_Leave(object sender, EventArgs e)
        {
            if (txtUsuario.Text != "")
            {
                lblNomeUsuario.Text = "";
                lblNomeUsuario.Location = new Point(155, 80);
                lblNomeUsuario.BackColor = Color.Transparent;
                lblNomeUsuario.AutoSize = true;
                lblNomeUsuario.ForeColor = Color.Blue;
                lblNomeUsuario.Font = new Font(lblNomeUsuario.Font, FontStyle.Bold);
                pictureBox1.Controls.Add(lblNomeUsuario);
                usuario.RetornaRegistro(Convert.ToInt32(txtUsuario.Text));
                lblNomeUsuario.Text = usuario.NomeUsuario;
            }
        }

        private void frmSenha_KeyPress(object sender, KeyPressEventArgs e)
        {
            if (e.KeyChar == 13)
            {
                e.Handled = true;
                SendKeys.Send("{TAB}");
;            }
        }

        private void txtUsuario_KeyPress(object sender, KeyPressEventArgs e)
        {
            if (!char.IsDigit(e.KeyChar) && e.KeyChar != 08)
            {
                e.Handled = true;
                Geral.TelaAviso("Campo usuário só aceita valores númericos \nFJSJHFH FASF ASDASD", "ERRO", this, Belemtech.TipoAviso.OK, Belemtech.ImagemAviso.Error);
            }
        }

        private void frmSenha_FormClosing(object sender, FormClosingEventArgs e)
        {
            if (_xResposta != 1)
            {
                Application.Exit();
            }
        }
    }
}
